### PR TITLE
Add more tests

### LINKED ISSUE #3: Add more tests
- Add tests for filter, mapping, transformation...
- Add tests for pipelines
- Add test with a filter that combine two pipelines
- Add a test fixture that starts oteapi-services

### PR DESCRIPTION
**Pull request by CasperWA**
Closes #3 

Note, this currently builds on the commits in #26 and is hence blocked by it.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -59,7 +59,7 @@ repos:
     - id: pylint-tests
       name: pylint - tests
       entry: pylint
-      args: ["--rcfile=pyproject.toml", "--extension-pkg-whitelist='pydantic'", "--disable=C0415,W0621"]
+      args: ["--rcfile=pyproject.toml", "--extension-pkg-whitelist='pydantic'", "--disable=C0415,W0621,duplicate-code"]
```
Comment (jesper-friis) [inline .pre-commit-config.yaml:62]: same here

File: tests/conftest.py [hunk 2]
```diff
@@ -11,14 +12,28 @@
...
+            Optional[OTEClient],
+        ],
+        None,
+    ]
+
+
+class ResourceType(str, Enum):
     """Enumeration of resource types."""
 
     DATARESOURCE = "dataresource"
```
Comment (jesper-friis) [inline tests/conftest.py:34]: Isn't FUNCTIONmissing here?

Comment (jesper-friis) [inline tests/conftest.py:34]: Isn't FUNCTIONmissing here?

Comment (jesper-friis) [inline tests/conftest.py:34]: Isn't FUNCTIONmissing here?

Comment (CasperWA) [reply tests/conftest.py:34]: It will be implemented in #30

**Pull request merged**